src/test/java/guitests/UIEventTests.java

```diff
@@ -66,4 +75,27 @@ public void panelClickedTest() {
         click("#dummy/dummy_col0_filterTextField");
         assertEquals(1, eventTestCount);
     }
+    
+    @Test
+    public void defaultRepoSwitchedTest() {
+        UI.events.registerEvent((PrimaryRepoChangedEventHandler) e -> UIEventTests.increaseEventTestCount());
+        UI.events.registerEvent((PrimaryRepoChangedEventHandler) e -> UIEventTests.getEventRepoId(e));
+        resetEventTestCount();
+        press(KeyCode.CONTROL).press(KeyCode.R).release(KeyCode.R).release(KeyCode.CONTROL);
+        assertEquals(1, eventTestCount);
+        resetEventTestCount();
+
+        // Test with multiple repositories
+        ComboBox<String> comboBox = find("#repositorySelector");
+        doubleClick(comboBox);
+        doubleClick();
+        type("dummy3/dummy3");
+        push(KeyCode.ENTER);
+        click("#dummy/dummy_col0_filterTextField");
+        resetEventTestCount();
+        press(KeyCode.CONTROL).press(KeyCode.R).release(KeyCode.R).release(KeyCode.CONTROL);
+        assertEquals(1, eventTestCount);
+        press(KeyCode.CONTROL).press(KeyCode.R).release(KeyCode.R).release(KeyCode.CONTROL);
+        assertEquals("dummy3/dummy3", defaultRepoId);
```

Comment:
Repo count 0 is not really valid to test for i.e. we don't have a mechanism to start the UI with zero repos opened. Repo count 1 is a valid case, though.